packages/flutter_tools/lib/src/asset.dart

```diff
@@ -644,7 +644,10 @@ class ManifestAssetBundle implements AssetBundle {
     final List<_Asset> sortedKeys = jsonEntries.keys.toList()
         ..sort((_Asset left, _Asset right) => left.entryUri.path.compareTo(right.entryUri.path));
     for (final _Asset main in sortedKeys) {
-      jsonObject[main.entryUri.path] = jsonEntries[main]!;
+      jsonObject[Uri.decodeFull(main.entryUri.path)] =
```

Comment:
Can you break this line into individual lines with named variables (to help readability) (please correct any var names that are wrong):

```dart
      final String decodedEntryUri = Uri.decodeFull(main.entryUri.path);
      final List<String> rawVariants = jsonEntries[main]!;
      final List<String> decodedVariants = rawVariants.map(Uri.decodeFull).toList();
      jsonObject[decodedEntryUri] = decodedVariants;
```

Reply:
Good call. I was little overzealous in avoiding variables that are only used once.